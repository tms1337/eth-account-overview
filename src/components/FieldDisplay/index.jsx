import React from "react";

const FieldDisplay = ({ value, icon }) => {
  return (
    <>
      {icon && <img src={icon} alt="Field icon" />}
      <div>{value}</div>
    </>
  );
};

export default FieldDisplay;
