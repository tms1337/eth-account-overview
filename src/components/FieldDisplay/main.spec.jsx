import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import FieldDisplay from ".";

describe("FieldDisplay", () => {
  it("renders snapshot", () => {
    const component = renderer.create(<FieldDisplay />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has correct float value", () => {
    const value = 12.5;
    const component = mount(<FieldDisplay value={value} />);

    expect(component.contains(value)).toBe(true);
  });

  it("has correct int value", () => {
    const value = 1337;
    const component = mount(<FieldDisplay value={value} />);

    expect(component.contains(value)).toBe(true);
  });

  it("has correct text value", () => {
    const value = "Different value";
    const component = mount(<FieldDisplay value={value} />);

    expect(component.contains(value)).toBe(true);
  });

  it("shows icon if passed", () => {
    const icon =
      "https://cdn4.iconfinder.com/data/icons/cryptocoins/227/ETH-512.png";
    const component = mount(<FieldDisplay icon={icon} />);

    expect(component.find(`img[src="${icon}"]`).length).toBe(1);
  });

  it("does not show icon if not passed", () => {
    const component = mount(<FieldDisplay />);

    expect(component.find("img").length).toBe(0);
  });
});
