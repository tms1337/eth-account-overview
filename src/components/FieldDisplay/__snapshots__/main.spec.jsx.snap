// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldDisplay renders snapshot 1`] = `<div />`;
